src/lib/PermissionStore.ts

```diff
@@ -0,0 +1,165 @@
+export interface Permission {
+    origin: string;
+    allowsAll: boolean;
+    addresses: string[];
+}
+
+export class PermissionStore {
+    public static readonly ALL_ADDRESSES = 'all-addresses';
+
+    public static readonly DB_VERSION = 1;
+    public static readonly DB_NAME = 'nimiq-permissions';
+    public static readonly DB_STORE_NAME = 'permissions';
+
+    public static INDEXEDDB_IMPLEMENTATION = window.indexedDB;
+
+    private static instance: PermissionStore;
+    private dbPromise: Promise<IDBDatabase>|null;
+    private indexedDB: IDBFactory;
+
+    public static get Instance() {
+        this.instance = this.instance || new PermissionStore();
+        return this.instance;
+    }
+
+    private constructor() {
+        this.dbPromise = null;
+        this.indexedDB = PermissionStore.INDEXEDDB_IMPLEMENTATION;
+    }
+
+    public allowByOrigin(origin: string, addresses: string|string[]): Promise<string> {
```

Comment:
I think we should leverage the advantages of a typed language and use `Nimiq.Address` here.

Reply:
I just don't know how to best handle returning `Permission` later (which should have `Nimiq.Address` in there as well).